src/Sylius/Bundle/AdminBundle/templates/product/grid/action/generate_variants.html.twig

```diff
@@ -2,7 +2,9 @@
 
 {% set path = options.link.url|default(path('sylius_admin_product_variant_generate', {'productId': options.product.id})) %}
 
-<a href="{{ path }}" class="btn">
-    {{ icon({ icon: 'arrows-split', class: 'icon icon-tabler' }) }}
-    {{ 'sylius.ui.generate'|trans }}
-</a>
+{% if options.product.options is not empty %}
```

Comment:
Include the path resolving within this if, there's no point to it when it's not used either way 🤷